src/Lecture2.hs

```diff
@@ -74,7 +75,16 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt ind l 
+  | ind < 0 = (Nothing, l)
+  | otherwise = go 0 [] l
+  where
+    go :: Int -> [a] -> [a] -> (Maybe a, [a])
+    go _ acc [] = (Nothing, reverse acc)
+    go i acc (x:xs)
+      | i == ind = (Just x, reverse acc ++ xs)
+      | otherwise = go (i + 1) (x:acc) xs
```

Comment:
That's a great solution! 👏🏻 
However, adding elements to the end of the list and calling `reverse` is expensive. It's possible to implement this function optimally 🙂